### ISSUE TITLE
ocaml.lang.best-practice.ifs.ocamllint-useless-else

### ISSUE THREAD

**Issue by mjambon** [returntocorp/sgrep-rules#1285]
**Describe the bug**
A clear and concise description of what the bug is and for which rule. Is the rule producing a false positive? Is it incorrect or missing something?

The example on the [landing page for the rule](https://semgrep.dev/r/ocaml.lang.best-practice.ifs.ocamllint-useless-else) makes no sense because the variable `a` can't be both a bool and a string:
```
let test a =
  (* ruleid:ocamllint-useless-else *)
  if a
  then print_string a
  else ()
```

The example should be something like:

```
let test a =
  (* ruleid:ocamllint-useless-else *)
  if cond () then
    print_string a
  else
    ()
```

Same for the example for the other rule that follows (`ocamllint-backwards-if`). Should be something like:

```
let test2 a =
  (* ruleid:ocamllint-backwards-if *)
  if cond () then
    ()
  else
    print_string a
```

**Priority**
How important is this to you?
- [ ] P0: blocking me from making progress
- [ ] P1: this will block me in the near future
- [x] P2: annoying but not blocking me
